packages/framework-provider-azure-infrastructure/src/infrastructure/index.ts

```diff
@@ -1,39 +1,76 @@
 import { BoosterConfig, Logger } from '@boostercloud/framework-types'
-import { ApplicationStackBuilder } from './stacks/application-stack'
-import {
-  azureCredentials,
-  createResourceGroup,
-  createResourceGroupName,
-  createResourceManagementClient,
-  createWebSiteManagementClient,
-} from './setup'
+import { App } from 'cdktf'
+import { AzureStack } from './azure-stack'
+import { ZipResource } from './types/zip-resource'
+import { FunctionZip } from './helper/function-zip'
+import { createFunctionResourceGroupName, createResourceGroupName, renderToFile } from './helper/utils'
+import { runCommand } from '@boostercloud/framework-common-helpers'
+import * as ckdtfTemplate from './templates/cdktf'
+
+export const synth = (configuration: BoosterConfig, logger: Logger): Promise<void> => synthApp(logger, configuration)
 
 export const deploy = (configuration: BoosterConfig, logger: Logger): Promise<void> => deployApp(logger, configuration)
 
 export const nuke = (configuration: BoosterConfig, logger: Logger): Promise<void> => nukeApp(logger, configuration)
 
+/**
+ * Synth the application for Azure
+ */
+async function synthApp(logger: Logger, config: BoosterConfig): Promise<void> {
+  await terraformSynth(logger, config)
+}
+
+async function uploadFile(logger: Logger, config: BoosterConfig, zipResource: ZipResource): Promise<void> {
+  logger.info('Uploading zip file')
+  const resourceGroupName = createResourceGroupName(config)
+  const functionAppName = createFunctionResourceGroupName(resourceGroupName)
+  await FunctionZip.deployZip(functionAppName, resourceGroupName, zipResource)
+  logger.info('Zip file uploaded')
+}
+
 /**
  * Deploys the application in Azure
  */
 async function deployApp(logger: Logger, config: BoosterConfig): Promise<void> {
-  const credentials = await azureCredentials()
-  const [resourceManagementClient, webSiteManagementClient] = await Promise.all([
-    createResourceManagementClient(credentials),
-    createWebSiteManagementClient(credentials),
-  ])
-  const resourceGroupName = createResourceGroupName(config)
-  await createResourceGroup(resourceGroupName, resourceManagementClient)
-  const applicationBuilder = new ApplicationStackBuilder(config)
-  await applicationBuilder.buildOn(logger, resourceManagementClient, webSiteManagementClient, resourceGroupName)
+  logger.info(`Deploying app ${config.appName}`)
+  const zipResource = await terraformSynth(logger, config)
+  const command = await runCommand(process.cwd(), 'cdktf deploy --auto-approve')
+  if (command?.stdout.includes('non-zero exit code')) {
```

Comment:
Why not just checking the exit code? I believe its more standard

Reply:
yeah, let me fix it